obkvrpc/compressor.go

```diff
@@ -0,0 +1,156 @@
+	}
+	return bytes.NewBuffer(dstBytes), nil
+}
+
+func (d *SnappyDecompressor) Decompress(src *bytes.Buffer, originLen int32) (*bytes.Buffer, error) {
+	srcBytes := src.Bytes()
+	dstBytes, err := snappy.Decode(nil, srcBytes)
+	if err != nil {
+		return nil, err
+	}
+	if int32(len(dstBytes)) != originLen {
+		return nil, errors.New(fmt.Sprintf("SnappyDecompressor length of decompress content is not expected: actual-%d, expected-%d", int32(len(dstBytes)), originLen))
+	}
+	return bytes.NewBuffer(dstBytes), nil
+}
+
+func (d *ZlibDecompressor) Decompress(src *bytes.Buffer, originLen int32) (*bytes.Buffer, error) {
+	// 1. create zlib reader
+	zlibReader, err := zlib.NewReader(src)
+	if err != nil {
+		return nil, err
+	}
+
+	defer func(zlibReader io.ReadCloser) {
+		err := zlibReader.Close()
+		if err != nil {
+			log.Warn(fmt.Sprintf("fail to close the zlibReader, errMsg: %s", err.Error()))
+		}
+	}(zlibReader)
+
+	// 2. create buffer
+	dstBuffer := new(bytes.Buffer)
+	// 3. write the decompress data to dstBuffer
+	n, err := io.Copy(dstBuffer, zlibReader)
+	if err != nil {
+		return nil, err
+	}
+	if int32(n) != originLen {
+		return nil, errors.New(fmt.Sprintf("ZlibDecompressor:length of decompress content is not expected: actual-%d, expected-%d", n, originLen))
+	}
+	return dstBuffer, nil
+}
+
+func (d *ZStdDecompressor) Decompress(src *bytes.Buffer, originLen int32) (*bytes.Buffer, error) {
+	zstdReader, err := zstd.NewReader(src)
+	if err != nil {
+		return nil, err
+	}
+	defer zstdReader.Close()
+	dstBuffer := new(bytes.Buffer)
+	n, err := io.Copy(dstBuffer, zstdReader)
+	if err != nil {
+		return nil, err
+	}
+	if int32(n) != originLen {
+		return nil, errors.New(fmt.Sprintf("ZStdDecompressor: length of decompress content is not expected: actual-%d, expected-%d", n, originLen))
+	}
+	return dstBuffer, nil
+}
+
+func getDecompressor(compressType protocol.ObCompressType) (DeCompressor, error) {
+	switch compressType {
+	case protocol.ObCompressTypeLZ4:
+		return &Lz4Decompressor{}, nil
+	case protocol.ObCompressTypeSnappy:
+		return &SnappyDecompressor{}, nil
+	case protocol.ObCompressTypeZlib:
+		return &ZlibDecompressor{}, nil
+	case protocol.ObCompressTypeZstd:
+		return &ZStdDecompressor{}, nil
+	case protocol.ObCompressTypeZstd_138:
+		return &ZStdDecompressor{}, nil
+	case protocol.ObCompressTypeLZ4_191:
+		return &Lz4Decompressor{}, nil
+	case protocol.ObCompressTypeNone:
+		return &NoneDecompressor{}, nil
+	default:
+		return nil, errors.New("unknown compression algorithm")
+	}
+}
+
+func convertCompressTypeToString(compressType protocol.ObCompressType) string {
+	switch compressType {
+	case protocol.ObCompressTypeLZ4:
+		return "lz4"
+	case protocol.ObCompressTypeSnappy:
+		return "snappy"
+	case protocol.ObCompressTypeZlib:
+		return "zlib"
+	case protocol.ObCompressTypeZstd:
+		return "zStd"
+	case protocol.ObCompressTypeNone:
+		return "none"
+	case protocol.ObCompressTypeZstd_138:
+		return "zstd_1.3.8"
```

Comment:
It seems different from 
``` source
    return "zStd"
```
Keep _zstd_ written in the same way

Reply:
I think it return the "zstd_1.3.8" and "zstd" may be a better way to classify which compress func observer used

Reply:
We should use _zStd_1.3.8_ in line 150 or _zstd_  in line 146.
Keep "s" in the same format.

Reply:
done, use 'zstd' to keep the same format